src/mixins/shallowPropertyComparisonMixin.ts

```diff
@@ -1,11 +1,13 @@
 import { entries } from 'dojo-shim/object';
 import { WidgetProperties } from './../interfaces';
+import { deepAssign } from 'dojo-core/lang';
 
 /**
  * Interface for `diffProperties`
  */
 export interface ShallowPropertyComparisonMixin {
-	diffProperties<T>(previousProperties: T): string[];
+	diffProperties<T>(previousProperties: T, newProperties: T): string[];
```

Comment:
Do you think we should extract these to a separate generic API that `shallowPropertyComparisonMixin` can implement and `WidgetMixin` can also extend.

If we are saying for any mixin that provides comparison functionality will need to override these function?